dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-kyuubi/src/main/java/org/apache/dolphinscheduler/plugin/datasource/kyuubi/param/KyuubiDataSourceProcessor.java

```diff
@@ -0,0 +1,147 @@
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.datasource.kyuubi.param;
+
+import org.apache.dolphinscheduler.common.constants.Constants;
+import org.apache.dolphinscheduler.common.constants.DataSourceConstants;
+import org.apache.dolphinscheduler.common.utils.JSONUtils;
+import org.apache.dolphinscheduler.plugin.datasource.api.datasource.AbstractDataSourceProcessor;
+import org.apache.dolphinscheduler.plugin.datasource.api.datasource.BaseDataSourceParamDTO;
+import org.apache.dolphinscheduler.plugin.datasource.api.datasource.DataSourceProcessor;
+import org.apache.dolphinscheduler.plugin.datasource.api.utils.PasswordUtils;
+import org.apache.dolphinscheduler.spi.datasource.BaseConnectionParam;
+import org.apache.dolphinscheduler.spi.datasource.ConnectionParam;
+import org.apache.dolphinscheduler.spi.enums.DbType;
+
+import org.apache.commons.collections4.MapUtils;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.auto.service.AutoService;
+
+@AutoService(DataSourceProcessor.class)
+public class KyuubiDataSourceProcessor extends AbstractDataSourceProcessor {
+
+    @Override
+    public BaseDataSourceParamDTO castDatasourceParamDTO(String paramJson) {
+        return JSONUtils.parseObject(paramJson, KyuubiDataSourceParamDTO.class);
+    }
+
+    @Override
+    public BaseDataSourceParamDTO createDatasourceParamDTO(String connectionJson) {
+        KyuubiDataSourceParamDTO kyuubiDataSourceParamDTO = new KyuubiDataSourceParamDTO();
+        KyuubiConnectionParam kyuubiConnectionParam = (KyuubiConnectionParam) createConnectionParams(connectionJson);
+        kyuubiDataSourceParamDTO.setDatabase(kyuubiConnectionParam.getDatabase());
+        kyuubiDataSourceParamDTO.setUserName(kyuubiConnectionParam.getUser());
+        kyuubiDataSourceParamDTO.setOther(kyuubiConnectionParam.getOther());
+
+        String[] tmpArray = kyuubiConnectionParam.getAddress().split(Constants.DOUBLE_SLASH);
+        StringBuilder hosts = new StringBuilder();
+        String[] hostPortArray = tmpArray[tmpArray.length - 1].split(Constants.COMMA);
+        for (String hostPort : hostPortArray) {
+            hosts.append(hostPort.split(Constants.COLON)[0]).append(Constants.COMMA);
+        }
+        hosts.deleteCharAt(hosts.length() - 1);
+        kyuubiDataSourceParamDTO.setHost(hosts.toString());
+        kyuubiDataSourceParamDTO.setPort(Integer.parseInt(hostPortArray[0].split(Constants.COLON)[1]));
+
+        return kyuubiDataSourceParamDTO;
+    }
+
+    @Override
+    public BaseConnectionParam createConnectionParams(BaseDataSourceParamDTO datasourceParam) {
+        KyuubiDataSourceParamDTO kyuubiParam = (KyuubiDataSourceParamDTO) datasourceParam;
+        StringBuilder address = new StringBuilder();
+        address.append(DataSourceConstants.JDBC_HIVE_2);
```

Comment:
`jdbc:kyuubi://` is recommended, of course, `jdbc:hive2://` is supported as well

Reply:
changed to jdbc:kyuubi://